pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -55,28 +56,53 @@ type AmbientIndex struct {
 	// byPod indexes by Pod IP address.
 	byPod map[string]*model.WorkloadInfo
 
+	services map[string]*workloadapi.Service
+
 	// Map of ServiceAccount -> IP
 	// TODO: currently, this is derived from pods. To be agnostic to the implementation,
 	// we should actually be looking at Gateway.status.addresses.
 	// This may be an external address (possibly even a DNS name we need to resolve), an arbitrary IP,
 	// or a reference to a service.
 	// If its a reference to a Service then we can find the underlying pods in that service, as an optimization.
-	waypoints map[model.WaypointScope]sets.String
+	waypoints map[model.WaypointScope]string
 
 	// serviceVipIndex maintains an index of VIP -> Service
 	serviceVipIndex *kclient.Index[*v1.Service, string]
 }
 
 // Lookup finds a given IP address.
-func (a *AmbientIndex) Lookup(ip string) []*model.WorkloadInfo {
+func (a *AmbientIndex) Lookup(ip string) []*model.AddressInfo {
 	a.mu.RLock()
 	defer a.mu.RUnlock()
 	// First look at pod...
 	if p, f := a.byPod[ip]; f {
-		return []*model.WorkloadInfo{p}
+		addr := &workloadapi.Address{
+			Type: &workloadapi.Address_Workload{
+				Workload: p.Workload,
+			},
+		}
+
+		return []*model.AddressInfo{&model.AddressInfo{Address: addr}}
 	}
 	// Fallback to service. Note: these IP ranges should be non-overlapping
-	return a.byService[ip]
+	res := []*model.AddressInfo{}
+	for _, wl := range a.byService[ip] {
+		addr := &workloadapi.Address{
+			Type: &workloadapi.Address_Workload{
+				Workload: wl.Workload,
+			},
+		}
+		res = append(res, &model.AddressInfo{Address: addr})
```

Comment:
i would expect this to return the new service type, yes? the whole point of the address proto is to wrap and allow lookups on either type via network/ip